packages/lab/src/dialog/DialogTitle.tsx

```diff
@@ -13,20 +16,37 @@ import dialogTitleCss from "./DialogTitle.css";
 
 const withBaseName = makePrefixer("saltDialogTitle");
 
-interface DialogTitleProps extends ComponentPropsWithoutRef<"h2"> {
+interface DialogTitleProps {
   /**
    * The status of the Dialog
-   * */
+   */
   status?: ValidationStatus | undefined;
   /**
-   * Displays the accent bar in the Dialog Title
-   * */
+   * Displays the accent bar in the Dialog Title */
   disableAccent?: boolean;
+  /**
+   * Displays the Dialog Title in a H2 component
+   */
+  title?: ReactNode;
+  /**
+   * Displays the Dialog Subtitle in a Label component
+   **/
+  subtitle?: ReactNode;
+
+  children?: ReactNode;
+
+  className?: string;
+  /**
+   * Displays the Dialog Subtitle in a Label component
+   * */
+  id?: string;
 }
 
 export const DialogTitle = ({
   children,
   className,
+  title,
+  subtitle,
```

Comment:
Passing these down as props was chosen over the composable api to ensure the formatting for Dialog Title and subtitle would be consistent across all dialog components